src/ymfm.h

```diff
@@ -244,7 +244,7 @@ inline int16_t roundtrip_fp(int32_t value)
 
 	// apply the shift back and forth to zero out bits that are lost
 	exponent -= 1;
-	return (value >> exponent) << exponent;
+	return uint32_t(value >> exponent) << exponent;
```

Comment:
This doesn't make sense. The std::max above guarantees the exponent is >= 1, so there can be no possibility of negative shift. What warning does the uint32_t cast fix? I'd honestly prefer an int16_t() cast on the way out since we're explicitly truncating to 16 bits.